packages/backend/src/services/RenameService/rename.ts

```diff
@@ -0,0 +1,666 @@
+import {
```

Comment:
This is where most of the `rename` logic is , all using static methods and factories